### PR TITLE
Issue 4938: Merge branch 'feature-4938-dr-tools-base-case' into 'master' 

### LINKED ISSUE #4938: DR - Ability to migrate segment metadata to a new deployment when data is fully flushed to LTS.
**Problem description**
Pravega administrator should be able to migrate data from existing Pravega deployment to a new Pravega deployment when all segments in existing deployments are fully flushed to LTS. 

**Problem location**
Segment store
Admin tool

**Suggestions for an improvement**
Implement/Enhance DebugSegmentStore that enumerates segments on existing LTS and populates the metadata in new deployment.

### PR DESCRIPTION
**Pull request by ManishKumarKeshri**
**Change log description**  
- Integration of Data recovery tool feature branch changes into the master  
- Feature branch change log description:
       - Listing segments from storage - `Iterator` for different kind of storages.
       - Data Recovery - Introduction of `DebugSegmentContainer` to `register` segments. Recovery logic for segments.
       - Tests: Unit tests, integration tests to test data recovery & segments' listing.
       - DR CLI Commands - CLI commands execution code along with tests.
 
**Purpose of the change**  
  Fixes #4938. Adds data recovery feature.

**What the code does**  
- Executes DR CLI commands when called to execute. 
- Introduces `DebugSegmentContainer` to register register. `ContainerRecoveryUtils` holds various methods required for recovery. 
- Introduces `Iterator` to list the segments in storage.

**How to verify it**  
- Build shall pass.
- DR CLI Commands should be able to list segments and recover data when used.

### CODE REVIEW

File: config/admin-cli.properties [hunk 1]
```diff
@@ -19,11 +19,43 @@ cli.controller.grpc.uri=tcp://localhost:9090
 cli.store.metadata.backend=segmentstore
 
 # General configuration for the Pravega cluster at hand.
-pravegaservice.cluster.name=pravega/pravega
+pravegaservice.clusterName=pravega/pravega
```
Comment (RaulGracia) [inline config/admin-cli.properties:22]: @ManishKumarKeshri please, use the existing name structure, as I think that the property name used in this PR is from the old naming structure. @ravisharda is that correct?

**Comment by RaulGracia** [comment]
@ManishKumarKeshri PR #5250 has been merged, which also touches the CLI package. Please, do a final iteration of test to check that everything works fine in this PR after merging these changes. cc/ @anirudhkovuru.

File: config/admin-cli.properties (hunk 1 above)
Comment (ManishKumarKeshri) [reply config/admin-cli.properties:24]: It should be `clusterName` as mentioned here: https://github.com/pravega/pravega/blob/b5a809388a650e95ec2cd901a2b2f92af47b7081/segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceConfig.java#L54

Comment (RaulGracia) [reply config/admin-cli.properties:24]: Got it, missed that one.

**Review by sachin-j-joshi — APPROVED** [review]
LGTM

**Comment by ManishKumarKeshri** [comment]
@RaulGracia Final iteration of testing DR CLI commands is successful.

**Pull request merged**